java/src/com/facebook/watchman/WatchmanClientImpl.java

```diff
@@ -71,12 +72,12 @@ public Boolean get() {
 
   @VisibleForTesting
   WatchmanClientImpl(
-      Supplier<Map<String, Object>> inputSupplier,
-      OutputStream outputStream,
+      Callable<Map<String, Object>> incomingMessageGetter,
```

Comment:
@styurin , I know you didn't like the name inputCallable or inputMessageCallable (and I agree with you), so I'm trying out this, instead:  `incomingMessageGetter` for the Callable that gives you messages from watchman, and `outgoingMessageStream` for the OutputStream that sends serialized messages to wachman.  Better?  Or do you have some other suggestion?